pkg/controller/secret/secret_controller.go

```diff
@@ -362,168 +380,16 @@ func updateAlertManagerConfig(r *ReconcileSecret, request *reconcile.Request, am
-	}
-
-	// Create a route for the new Watchdog receiver.
-	wdroute := &alertmanager.Route{
-		Receiver:       "watchdog",
-		RepeatInterval: "5m",
-		Match:          map[string]string{"alertname": "Watchdog"},
-	}
-
-	// Insert the Route for the Watchdog Receiver.
-	routeabsent := true
-	log.Info("DEBUG: Checking for watchdog related routes")
-	for i, route := range amconfig.Route.Routes {
-		log.Info("DEBUG: Found Route", "receiver", route.Receiver)
-		switch route.Receiver {
-		case "watchdog":
-			log.Info("DEBUG: Overwriting Watchdog Route", "receiver", route.Receiver)
-			amconfig.Route.Routes[i] = wdroute
-			routeabsent = false
-		case "null":
-			// Remove null route, since the watchdog route replaces it.
-			log.Info("DEBUG: Deleting Route for Receiver:", route.Receiver)
-			amconfig.Route.Routes = removeFromRoutes(amconfig.Route.Routes, i)
-		default:
-			log.Info("DEBUG: Skipping route for receiver", "name", route.Receiver)
-		}
-	}
-
-	// Create Route for Watchdog Receiver if it doesn't already exist.
-	if routeabsent {
-		log.Info("DEBUG: Route for Watchdog Receiver is absent. Creating new Route.")
-		amconfig.Route.Routes = append(amconfig.Route.Routes, wdroute)
-	}
-
-	// Update the default route to point to our new receiver.
-	amconfig.Route.Receiver = "watchdog"
-
-}
-
-// removeFromReceivers removes the specified index from a slice of Receivers.
-func removeFromReceivers(r []*alertmanager.Receiver, i int) []*alertmanager.Receiver {
-	r = append(r[:i], r[i+1:]...)
-	return r
-}
-
-// removeFromRoutes removes the specified index from a slice of Routes.
-func removeFromRoutes(r []*alertmanager.Route, i int) []*alertmanager.Route {
-	r = append(r[:i], r[i+1:]...)
-	return r
-}
-
-// removeConfigFromAlertManager removes a Receiver config and the associated Route from Alertmanager.
-// The changes are kept in memory until committed using function updateAlertManagerConfig().
-func removeConfigFromAlertManager(r *ReconcileSecret, request *reconcile.Request, amconfig *alertmanager.Config, receivername string) {
-	log.Info("DEBUG: Checking for receiver in Alertmanager config", "name", receivername)
-
-	nullrouteabsent := true
-	for i, receiver := range amconfig.Receivers {
-		if receiver.Name == receivername {
-			log.Info("DEBUG: Deleting receiver", "name", receiver.Name)
-			amconfig.Receivers = removeFromReceivers(amconfig.Receivers, i)
-		}
-		if receiver.Name == "null" {
-			nullrouteabsent = false
-		}
-	}
-	for i, route := range amconfig.Route.Routes {
-		if route.Receiver == receivername {
-			log.Info("DEBUG: Deleting Route", "receiver", route.Receiver)
-			amconfig.Route.Routes = removeFromRoutes(amconfig.Route.Routes, i)
-		}
-	}
-
-	if nullrouteabsent {
-		nullreceiver := &alertmanager.Receiver{Name: "null"}
-		amconfig.Receivers = append(amconfig.Receivers, nullreceiver)
-	}
-
-	// If watchdog is being removed, put the system default route and receiver back into place.
```

Comment:
Has this been tested if DMS is removed? Does the default receiver go back in?
(this would be the case where someone adds `noalerts` to a cluster and the syncset removes the secret)

Reply:
The "null" route is now always set, it wasn't before.  This no longer tries to restore to system defaults.  It just has a bare bones route to "null" receiver.